core/app/services/spree/credit_cards/destroy.rb

```diff
@@ -0,0 +1,45 @@
+module Spree
+  module CreditCards
+    class Destroy
+      prepend Spree::ServiceModule::Base
+
+      def call(card:)
+        ApplicationRecord.transaction do
+          run :invalidate_payments
+          run :void_payments
+          run :destroy
+        end
+      end
+
+      protected
+
+      def invalidate_payments(card:)
+        if payment_scope(card).checkout.each(&:invalidate!)
+          success(card: card)
+        else
+          failure('invalidating payments failure')
+        end
+      end
+
+      def void_payments(card:)
+        if payment_scope(card).where.not(state: :checkout).each(&:void!)
+          success(card: card)
+        else
+          failure('voiding payments failure')
+        end
+      end
+
+      def destroy(card:)
+        if card.destroy!
```

Comment:
`destroy!` can raise exception, probably we want to use`destroy` to be able to present validation errors in case of a failure?